### PR TITLE
Add Support for Check Suites

### PR DESCRIPTION
**Pull request by wrecker**
~~NOTE: This PR includes changes from PR #1631~~ Cleaned up.

### CODE REVIEW

**Comment by wrecker** [comment]
The failures for Python 3.6 doesn't seem to be related to my change.. am i missing something?

**Comment by s-t-e-v-e-n-k** [comment]
It's lint again :-) I've merged your previous changes, please rebase.

**Comment by wrecker** [comment]
@s-t-e-v-e-n-k .. I don't understand whats causing these lint errors. 
`github/Workflow.pyi:5: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/Tag.pyi:3: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/Repository.pyi:8: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/PullRequest.pyi:4: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/GithubObject.pyi:3: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/Comparison.pyi:3: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/Commit.pyi:4: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/Commit.pyi:15: error: Name 'Commit' already defined (possibly by an import)
github/Branch.pyi:4: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?
github/MainClass.pyi:6: error: Module 'github.Commit' has no attribute 'Commit'; maybe "GitCommit"?`

**Comment by wrecker** [comment]
Ignore my last comment. Fixed the lint errors. I also setup a python3.6 environment using docker. I can now run `tox` locally. So my next PR should be better :)

File: github/Commit.pyi [hunk 1]
```diff
@@ -40,8 +41,13 @@ class Commit(CompletableGithubObject):
...
     @property
     def files(self) -> List[File]: ...
+    def get_check_suites(
+        self,
+        app_id: Union[_NotSetType, int],
+        check_name: Union[_NotSetType, str],
+    ) -> PaginatedList[CheckSuite]: ...
     def get_combined_status(self) -> CommitCombinedStatus: ...
-    def get_comments(self) -> PaginatedList[GitCommit]: ...
+    def get_comments(self) -> PaginatedList[CommitComment]: ...
```
Comment (s-t-e-v-e-n-k) [inline github/Commit.pyi:50]: This should probably be a separate change, but nice catch. :-)

**Comment by wrecker** [comment]
Sorry for the long silence.. had to deal with production issues at work and then took a short break. Will start revisiting this and other PRs this week.

**Comment by hawkins** [comment]
Very excited to see this, I would definitely use the checks api as soon as it ships. Any way I could lend a hand?

**Comment by dhruvmanila** [comment]
I wanted the Check Run API endpoint for my side project so I created all the endpoints present in https://docs.github.com/en/free-pro-team@latest/rest/reference/checks#check-runs which includes:
- Creat a check run
- Get a check run
- Update a check run
- List check run annotations
- List check runs for Git reference

The "List check runs in a check suit" should be implement from the `check-suite` endpoint.

All the Python files, stub files, test files and replay data files have been written, I will create a PR in a few days as I am not in town currently. I have also separated them out in logical commits like single object files commit, single endpoint files commit, test files and replay data files.

File: github/CheckSuite.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    This class represents check suites. The reference can be found here https://docs.github.com/en/rest/reference/checks#check-suites
+    """
+
+    def __repr__(self):
+        return self.get__repr__({"id": self._id.value, "url": self._url.value})
+
+    @property
+    def after(self):
+        """
+        :rtype: string
```
Comment (s-t-e-v-e-n-k) [inline github/CheckSuite.py:39]: Properties are 'type' not 'rtype'

File: github/CheckSuite.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    def url(self):
+        """
+        :rtype: string
+        """
+        self._completeIfNotSet(self._url)
+        return self._url.value
+
+    def rerequest(self):
+        """
+        :rtype: bool
```
Comment (s-t-e-v-e-n-k) [inline github/CheckSuite.py:166]: This actually calls a URL, so this needs more detail.

**Review by s-t-e-v-e-n-k — CHANGES_REQUESTED** [review]
Sorry for taking so long, just two more nits and this can be merged.

**Comment by sbesson** [comment]
At [OME](github.com/ome), we are making extensive use of PyGithub  in our own [library](github.com/ome/scc/) to filter PRs opened against all our organizational repository and create daily integration branches. First and foremost, thanks for the continued support of this Python bridge.

As we are assessing the transition to GitHub Actions for CI/CD, we realized it make extensive use of the Checks API (as opposed to the Status API used by Travis CI/Circle CI) for reporting the outcome of its check runs and suites. In case it's useful, we performed some testing of this branch and were able to access check-suites and their metadata.

Looking forward to see this feature (and the related https://github.com/PyGithub/PyGithub/pull/1727 from @dhruvmanila) landing in a PyGitHub release. Let us know if we can help with the integration.

**Comment by s-t-e-v-e-n-k** [comment]
@wrecker Thank you so much for your contribution, but I'm going to close this in favour of #1764, which incorporates your work.

**Pull request closed without merging**